### DISCUSSION TITLE
Upload files

### DISCUSSION THREAD

**Discussion by manson** [kamalshkeir/kago#9] [General]
Hi!
I have started my new small project and have decided to use Kago. It is going well for a while, but I have a small comment regarding images uploading to a user's profile. As I see, the file is being uploaded into the media folder under the original file's name. This is not good due to users may have files in a different coding (native languages) and at some point it may fail working with this (either uploading or downloading). Besides, I don't know how this shall behave if I upload different files to different profiles, with the same file's name.  This may overwrite one another.
The usual way is to autogenerate a new neutral name (like GUID) under which it is being saved into the file system, and probably save the original name in a database, so when downloading, to substitute generated name to original. I didn't check the uploading by using the API, but here - in admin panel the situation is like that.
If I find some more, will respond in discussions.

**Comment by kamalshkeir** [comment]
Hi Manson, glad to hear from you, great news for your project.
About uploading from admin, i agree with you, it will be better to do it like you describe it, i will add it to my list, and looking forwards for further feedbacks.
Btw, if you don't need the admin dashboard, consider using korm first, you can from it use the bus and kmux router which is the same as kago, i was working on implementing many2many and some fixes, it will be pushed today

**Reply by manson** [reply]
No, admin panel is needed at least on developer stage, and later as well.  
" kmux router which is the same as kago, i was working on implementing many2many and some fixes, it will be pushed today" - hope this will be available for kago as well, not only to kmux (as far as I remember now kago is a combining box for all other modules, right?). As far as I understand, you have incorporated a special logic to sql builder engine of korm for this purpose?

**Reply by manson** [reply]
I have implemented a simple SSR for my project by using VueJs and kago. UI project builds my html and libraries, copies it to kago project into templates and static folders and by using c.Html with templates variables binding results to user a fully functional SSR page. I like it very much how it works. So will work on it further.

**Reply by kamalshkeir** [reply]
`UI project builds my html and libraries, copies it to kago project into templates and static folders` and after that you embed them also in your binary https://github.com/kamalshkeir/kago#build-single-binary-with-all-static-and-html-files, and compress it using upx :) , you will have a single binary to deploy with all your static and templates

**Reply by manson** [reply]
>and after that you embed them.  

That's right :-)
